### DISCUSSION TITLE
How to  Externalize  the Camel Jbang console logs to the the External File?????

### DISCUSSION THREAD

**Discussion by kondetiramanakumar** [apache/camel-karavan#1350] [Q&A]
### Ask a question

How to  Externalize  the Camel Jbang console logs to the the External File?????

### Variant

None

### Container Management (if applicable)

None

### Operating System (if applicable)

None

### Version

None

**Comment by kondetiramanakumar** [comment]
I want to Externalise the Jbang Logs to the externalise file .

<img width="791" alt="image" src="https://github.com/user-attachments/assets/7df70f1f-8be6-4a52-bbc8-35d54caf6166">

**Comment by kondetiramanakumar** [comment]
@davsclaus  please help to provide your valuable Inputs???

**Comment by kondetiramanakumar** [comment]
@mgubaidullin Trust things are well with you since we last Interacted. Required your Inputs on this ticket. Your swift response would be highly appreciated.

**Comment by kondetiramanakumar** [comment]
FYI,
I am trying in the vscode Extension, I have added up my log4j.xml in the class path

<img width="504" alt="image" src="https://github.com/user-attachments/assets/0ff7b899-afd3-4611-a08b-77746827775a">

I want to push the camel jbang logs to the external file . any suggestions or any inputs????

**Comment by davsclaus** [comment]
See the ` --logging-config-path` option in camel-jbang run command

**Comment by kondetiramanakumar** [comment]
@davsclaus 

I used the below camel-jbang run command : I'm still not able to achieve exactly what I want.
jbang "-Dcamel.jbang.version=4.5.0" camel@apache/camel run * --logging-config-path="D:/Project-InsulationLayer/Architecture-Karavan/Base/Karavan/log4j2.xml" --verbose  > "D:/Project-InsulationLayer/Architecture-Karavan/Base/Karavan/logs/app.log".

By using the above command I am able to externalise the logs but not able to use the specified log4j2.xml file for logging. please help.

Firstly I am able to externalize the logs but I want to  tell the  JBang to use the specified Log4j2 configuration file (log4j2.xml) for logging. It directs Camel to configure logging based on this file?????. please help on this. required your value inputs?????

**Comment by kondetiramanakumar** [comment]
my Log4j2.xml

<?xml version="1.0" encoding="UTF-8"?>
<Configuration status="warn">
    <Properties>
        <Property name="basePath">D:/Project-InsulationLayer/Architecture-Karavan/Base/Karavan/logs</Property>
    </Properties>

    <Appenders>
        <RollingFile name="fileLogger"
            fileName="${basePath}/app.log"
            filePattern="${basePath}/app-%d{yyyy-MM-dd}.log">
            <PatternLayout>
                <pattern>[%-5level] %d{yyyy-MM-dd HH:mm:ss.SS} [%t] %c{1} - %msg%n</pattern>
            </PatternLayout>
            <Policies>
                <TimeBasedTriggeringPolicy interval="1" modulate="true" />
                <SizeBasedTriggeringPolicy size="10MB" />
            </Policies>
            <DefaultRolloverStrategy max="10">
                <Delete basePath="${basePath}" maxDepth="10">
                    <IfLastModified age="30d" />
                </Delete>
            </DefaultRolloverStrategy>
        </RollingFile>
        
        <Console name="consoleLogger">
            <PatternLayout>
                <pattern>[%-5level] %d{yyyy-MM-dd HH:mm:ss.SS} [%t] %c{1} - %msg%n</pattern>
            </PatternLayout>
        </Console>
    </Appenders>

    <Loggers>
        <Root level="info" additivity="false">
            <AppenderRef ref="fileLogger" />
            <AppenderRef ref="consoleLogger" />
        </Root>
        
        <Logger name="org.camel.karavan.demo.demo21062024" level="info" additivity="false">
            <AppenderRef ref="fileLogger"/>
            <AppenderRef ref="consoleLogger"/>
        </Logger>
    </Loggers>
</Configuration>

**Comment by kondetiramanakumar** [comment]
@davsclaus please help to provide your inputs for the above query????. Your swift response would be highly appreciated.
